extension/src/popup/views/Account/hooks/useGetAccountHistoryData.tsx

```diff
@@ -58,9 +71,18 @@ function useGetAccountHistoryData() {
           networkDetails: networkDetails,
           publicKey,
           fetchTokenDetails,
+          icons: cachedIcons,
```

Comment:
For account history icons, previously we would need to make a new lookup for every icon (regardless of if we had fetched it earlier in the iteration) because selectors only update on a new render. It wasn't possible to cache using redux

Now, I'm also passing along a temporary obj that caches newly fetched icons as we iterate. If we fetched (for ex) a USDC icon once, we don't need to look up the home domain's toml again the next time we come across the need for a USDC icon further down the loop.